### PR TITLE
feat(18788): Integrate Paypal SDK for basic Subscription payments

### PR DESCRIPTION
**Pull request by albaranau**
https://kontur.fibery.io/Tasks/Task/FE-Integrate-Paypal-SDK-for-subscription-payments-18788

## Summary by CodeRabbit

- **New Features**
  - Introduced PayPal payment integration for subscription management.
  - Added a `SubscriptionSuccessModal` to display subscription success messages.

- **Improvements**
  - Wrapped key components (`GlobalModal`, `Router`, `PricingPage`) in a new `PaymentsProvider` for enhanced payment handling.
  - Added new error and success messages related to payment and subscription processes.

- **Bug Fixes**
  - Improved error messages for payment initialization issues.

### CODE REVIEW

File: src/features/subscriptions/components/PayPalButtonsGroup/PayPalButtonsGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      forceReRender={[activeSubscriptionId, activeBillingPlanId, billingPlanId]}
+      onInit={(data, actions) => {}}
+      onError={(err) => console.error('error from PayPal SDK:', err.toString())}
+      onApprove={(data, actions) => {
+        if (!actions.subscription) {
+          return Promise.reject(
+            'unexpected error from PayPal SDK: onApprove was called, but actions.subscription is undefined',
+          );
+        }
+        return actions.subscription.get().then(function (details) {
```
Comment (dqunbp) [inline src/features/subscriptions/components/PayPalButtonsGroup/PayPalButtonsGroup.tsx]: De we need details here?

File: src/features/subscriptions/components/PayPalButtonsGroup/PayPalButtonsGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      style={{
+        label: 'subscribe',
+        color: 'blue',
+        height: 38,
+        shape: 'rect',
+      }}
+      forceReRender={[activeSubscriptionId, activeBillingPlanId, billingPlanId]}
+      onInit={(data, actions) => {}}
+      onError={(err) => console.error('error from PayPal SDK:', err.toString())}
+      onApprove={(data, actions) => {
```
Comment (dqunbp) [inline src/features/subscriptions/components/PayPalButtonsGroup/PayPalButtonsGroup.tsx]: Can we use async/await within the callbacks?

Comment (albaranau) [reply src/features/subscriptions/components/PayPalButtonsGroup/PayPalButtonsGroup.tsx]: @dqunbp depends on the callback, some of them must return Promise (e.g. `onApprove` and `createSubscription`)

File: src/features/subscriptions/components/PayPalButtonsGroup/PayPalButtonsGroup.tsx (hunk 1 above)
Comment (albaranau) [reply src/features/subscriptions/components/PayPalButtonsGroup/PayPalButtonsGroup.tsx]: Fixed

File: src/features/subscriptions/components/PricingContent/PricingContent.tsx [hunk 3]
```diff
@@ -38,6 +40,29 @@ export function PricingContent() {
...
+          if (
+            cycle.billingMethods.find(
+              (billingMethod) =>
+                billingMethod.billingPlanId === currentSubscription.billingPlanId,
+            )
+          ) {
+            setCurrentBillingCycleID(cycle.id);
+            return;
+          }
+        }
```
Comment (Natallia-Harshunova) [inline src/features/subscriptions/components/PricingContent/PricingContent.tsx:62]: Maybe its better to het rid of some nesting here with something like this

```
    const currentPlan = config.plans.find((plan) => plan.id === currentSubscription?.id);
    if(!currentPlan) return;
    
    //rest of the code
```

**Review by Natallia-Harshunova — APPROVED** [review]

File: src/features/subscriptions/components/SubscriptionSuccessModal/SubscriptionSuccessModal.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+export default function SubscriptionSuccessModal({ onConfirm }: Props) {
+  return (
+    <ModalDialog
+      title={i18n.t('subscription.success_modal.title')}
+      onClose={() => {
+        onConfirm(null);
+      }}
+      footer={
+        <div className={s.buttonsRow}>
```
[the remark this answers is not in the document]
Comment (albaranau) [reply src/features/subscriptions/components/SubscriptionSuccessModal/SubscriptionSuccessModal.tsx:17]: Please check again whether this `<div>` tag has no children

File: src/features/subscriptions/components/PricingContent/PricingContent.tsx (hunk 3 above)
Comment (albaranau) [reply src/features/subscriptions/components/PricingContent/PricingContent.tsx:62]: Fixed

**Review by vkozio — APPROVED** [review]

File: src/features/subscriptions/components/SubscriptionSuccessModal/SubscriptionSuccessModal.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+type Props = {
+  onConfirm: (value?: never) => void;
+};
+
+export default function SubscriptionSuccessModal({ onConfirm }: Props) {
+  return (
+    <ModalDialog
+      title={i18n.t('subscription.success_modal.title')}
+      onClose={() => {
```
[the remark this answers is not in the document]
Comment (albaranau) [reply src/features/subscriptions/components/SubscriptionSuccessModal/SubscriptionSuccessModal.tsx:13]: How do you know there's no such key?

[the remark this answers is not in the document]
Comment (albaranau) [reply src/features/subscriptions/components/SubscriptionSuccessModal/SubscriptionSuccessModal.tsx:13]: What about `common.json` file?

[the remark this answers is not in the document]
Comment (albaranau) [reply src/features/subscriptions/components/SubscriptionSuccessModal/SubscriptionSuccessModal.tsx:13]: It does exist in `common.json` file

File: src/features/subscriptions/components/PayPalButtonsGroup/PayPalButtonsGroup.tsx (hunk 2 above)
Comment (albaranau) [reply src/features/subscriptions/components/PayPalButtonsGroup/PayPalButtonsGroup.tsx]: fixed

**Pull request merged**